src/THORChain/Signer.cpp

```diff
@@ -14,23 +14,56 @@
 using namespace TW;
 
 namespace TW::THORChain {
-const std::string TYPE_PREFIX_MSG_SEND = "thorchain/MsgSend";
+const std::string TYPE_PREFIX_MSG_SEND_THOR = "thorchain/MsgSend";
+const std::string TYPE_PREFIX_MSG_SEND_MAYA = "mayachain/MsgSend";
+const std::string FROM_ADDRESS = "fromAddress";
+const std::string TO_ADDRESS = "toAddress";
+const std::string MAYACHAIN = "mayachain";
+const std::string MAYA_ADDRESS = "maya1";
 
 Cosmos::Proto::SigningOutput Signer::sign(Cosmos::Proto::SigningInput& input) noexcept {
+    std::string TypePrefixMsgSend;
+    TWCoinType TWCoinType;
+
+    if(input.chain_id().find(MAYACHAIN) != std::string::npos){
+        TypePrefixMsgSend = TYPE_PREFIX_MSG_SEND_MAYA;
+        TWCoinType = TWCoinTypeMAYAChain;
+    } else{
+        TypePrefixMsgSend = TYPE_PREFIX_MSG_SEND_THOR;
+        TWCoinType = TWCoinTypeTHORChain;
+    }
```

Comment:
For this part, I remove the duplicate code and modify the THORChain one, as you suggest:  `It would be great if you reuse THORSwap existing code my making required changes there.`

Reply:
Yes but I mean duplicating `THORChain::Signer::signJSON` and `THORChain::Signer::sign` only as they are small methods. This would allow `THORChain::Signer` to separate MAYAChain and THORChain specific parts. You already have `MAYAChain::Signer`, and we can move the MAYAChain specific functionality from `THORChain/Signer.cpp` to `MAYAChain/Signer.cpp`

Reply:
Ok, I already re-added `MAYAChain/Signer.cpp`